src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/ListingService.java

```diff
@@ -164,30 +164,25 @@ public ListingData processListingHearingsRequest(ListingDetails listingDetails,
         List<ListingTypeItem> listingTypeItems = new ArrayList<>();
         if (isHearingTypeValid(listingData, hearingTypeItem)) {
             int hearingDateCollectionSize = hearingTypeItem.getValue().getHearingDateCollection().size();
-            for (int i = 0; i < hearingDateCollectionSize; i++) {
+            for (var i = 0; i < hearingDateCollectionSize; i++) {
                 log.info("EthosCaseRef Listing: " + caseData.getEthosCaseReference());
                 hearingTypeItem.getValue().getHearingNumber();
                 log.info("Hearing number: " + hearingTypeItem.getValue().getHearingNumber());
-                DateListedTypeItem dateListedTypeItem = hearingTypeItem.getValue().getHearingDateCollection().get(i);
+                var dateListedTypeItem = hearingTypeItem.getValue().getHearingDateCollection().get(i);
                 boolean isListingVenueValid = isListingVenueValid(listingData, dateListedTypeItem);
-                log.info("isListingVenueValid: " + isListingVenueValid);
-                if (!isListingVenueValid) {
-                    continue;
-                }
                 boolean isListingDateValid = isListingDateValid(listingData, dateListedTypeItem);
+                log.info("isListingVenueValid: " + isListingVenueValid);
                 log.info("isListingDateValid: " + isListingDateValid);
-                if (!isListingDateValid) {
-                    continue;
-                }
+                var isListingStatusValid = true;
                 if (!showAllHearingType(listingData)) {
-                    boolean isListingStatusValid = isListingStatusValid(dateListedTypeItem);
+                     isListingStatusValid = isListingStatusValid(dateListedTypeItem);
                     log.info("isListingStatusValid: " + isListingStatusValid);
-                    if (!isListingStatusValid) {
-                        continue;
-                    }
                 }
-                ListingTypeItem listingTypeItem = new ListingTypeItem();
-                ListingType listingType = ListingHelper.getListingTypeFromCaseData(
+                if (!isListingVenueValid || !isListingDateValid || !isListingStatusValid) {
```

Comment:
You're definitely right to refactor this code but I don't think this is the correct way to do it. The code is now less efficient. Previously if isListingVenueValid was false then it would stop processing the hearing. Now it will do all three checks before deciding to stop processing or not.

How about removing all the checks and combining them into a new method then simply having this in getListingTypeItems

```
if (!isListingValid(listingData, dateListedTypeItem)) {
   continue;
}
```

I think all the log messages can be removed.

Reply:
No, The OR (||) operator doesn't check the next condition if the condition before it does fail. We can definitely move the if logic to a different function but it still doesn't make any difference because if !isListingVenueValid is false then !isListingDateValid will not be checked. I don't think the efficiency of code is compromised.

Reply:
If isListingVenueValid is false then it was a waste of time calculating the value of isListingDateValid and isListingStatusValid in the code above because you will never check the value.